standalone-tools/wss-scan.sh

```diff
@@ -0,0 +1,133 @@
+#                export wss_apikey=$(APIKEY)
+#
+# Usage:         ./wss-scan.sh
+#
+# Starting Date: 10-13-2020
+# Modified Date: 10-21-2020
+###############################################################################################
+
+set -e
+
+java -version
+if [ "$?" != 0 ]
+then 
+  echo "Java has not been setup" 
+  exit 1
+fi
+
+if [ ! -f "wss-unified-agent.jar" ]
+then
+  # Download the WhiteSource Agent 
+  wget -q https://github.com/whitesource/unified-agent-distribution/releases/latest/download/wss-unified-agent.jar
+  # The version 20.9.2.1 has been tested and can be used if a specific version is required
+  #curl -LJO -sS https://github.com/whitesource/unified-agent-distribution/releases/download/v20.9.2.1/wss-unified-agent.jar
+fi
+
+# scan the config file for the user configurations
+# wss-scan.config has to be present in the same working directory as the script
+source wss-scan.config
+
+# change comma to whitespace
+gitRepos=${gitRepos//,/$'\n'} 
+
+basepath=$baseDirPath"/repos"
+
+echo "Cleaning up scan directories if already present"
+rm -rf $basepath
+
+mkdir -p $basepath
+ 
+
+# clone the desired Repos for scanning 
+for repo in $gitRepos
+do
+  echo "Cloning repo "$gitBasePath$repo
+  git clone "$gitBasePath$repo".git $basepath"/"$repo
+done
+
+echo -n > info.txt
+
+# scan the Repos using the WhiteSource Unified Agent
+for repo in $gitRepos
+do
+  repo_path=$basepath"/"$repo
+  if [ -d "$repo_path" ]
+  then
+    echo "Scanning repo: "$gitBasePath$repo " Project: " $repo 
+    java -jar wss-unified-agent.jar -c wss-unified-agent.config -d $repo_path -apiKey $wss_apikey -product ODFE -project $repo | grep "Project name" | sed 's/^.\{,41\}//' >> info.txt 2>&1 &
+  else
+    echo "Scanning failed for repo: "$gitBasePath$repo " Project: " $repo
+  fi
+done
+
+
+i=0
+# wait for the scannings to complete
+while ps ax | grep -wq [w]ss-unified-agent.jar
+do
+  echo "Scanning is still in progress"
+  sleep 60
+  ((i=i+1))
+  # break the loop after 70 mins
+  if [ "$i" -gt 70 ] 
+  then
+    break 
+  fi
+done
+echo "Scanning has completed"
+
+
+# mail function to send the scan details to the desired recepient 
+mail_format_func()
+{
+
+echo "<html><body><table border=1 cellspacing=0 cellpadding=3>" > output.md
+while IFS= read -r line
+do
+# setting comma as the delimiter
+
+  IFS=','
+  read -ra val <<< "$line"
+  echo "<tr>" >> output.md
+  for ln in "${val[@]}" 
+  do
+    echo "${ln//[[:space:]]/}"
+    echo "<td>"${ln//[[:space:]]/}"</td>" >> output.md
+  done
+  echo "</tr>" >> output.md
+done < info.txt
+echo "</table></body></html>" >> output.md
+
+}
+
+mail_format_func
+
+# remove the functionality for local mail 
+
+#if [ "${emailid}" != "" ]; then
+#{
+#echo "Sending mail"
+#echo -e "Content-Type: text/html; charset='utf-8'\r\nSubject: ODFE Vulnerability Scan details" |cat - output.md |sendmail -t ${emailid}
+#}
+#fi
```

Comment:
Don't check in commented code. Just remove it. Since the sending of email is no longer part of this script, move the `mail_format_func` into its own script and have the Github workflow invoke it directly.

Reply:
Removed the sendmail section.